ACL/xattr support for OpenBSD 5.7

Hi, I am trying to build Burp on OpenBSD but am having issues with rsync. I've installed the package librsync onto the server but ./config is still exiting with what looks like a librsync error. Do you have any suggestions on what I could try next ?

checking whether closedir returns void... no
checking for gettext in -lintl... no
checking for getpwnam in -lsun... no
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
checking for deflate in -lz... yes
checking crypt.h usability... no
checking crypt.h presence... no
checking for crypt.h... no
checking for crypt in -lcrypt... no
checking uthash.h usability... yes
checking uthash.h presence... yes
checking for uthash.h... yes
checking librsync.h usability... yes
checking librsync.h presence... yes
checking for librsync.h... yes
checking for rs_delta_begin in -lrsync... no
Please install librsync
# 


Thanks
Keith

I was using the master branch and have now switched to v1.3. I tried to compile librsync and was very happy to see the post from gkre as changing the ./configure line seems to have got passed librsync issue although it has raised another.

LDFLAGS="-L/usr/local/lib" ./configure

and

./configure --prefix=/usr/local --sbindir=/usr/local/sbin LDFLAGS="-L/usr/local/lib"

both result in the following..

configure: creating ./config.status
config.status: creating autoconf/Make.common
config.status: creating Makefile
config.status: creating manpages/Makefile
config.status: creating src/Makefile
config.status: creating src/config.h
config.status: executing libtool commands
config.status: executing po-directories commands
config.status: executing default commands
Doing make of dependencies
==>Entering directory /root/burp-1.3-master/src
xattr.c:263:25: error: sys/extattr.h: No such file or directory
xattr.c:264:21: error: libutil.h: No such file or directory
*** Error 1 in src (Makefile:334 'depend': @g++ -S -M -I/usr/local/include -fno-strict-aliasing -fno-exceptions -fno-rtti   -I. -I..  *.c >>...)
*** Error 1 in /root/burp-1.3-master (Makefile:160 'depend')

Configuration on Thu Jul 30 11:02:11 BST 2015:

   Host:                        x86_64-unknown-openbsd5.7 -- openbsd
   Burp version:                Burp 1.3.49
   Source code location:        .
   Install binaries:            /usr/local/sbin
   Install config files:        /usr/local/etc
   Man directory:               ${datarootdir}/man
   C Compiler:                  gcc
   C++ Compiler:                g++
   Compiler flags:               -g -O2 -fno-strict-aliasing -fno-exceptions -fno-rtti
   Linker flags:                 -L/usr/local/lib
   Libraries:                    -lintl
   Static:

   ncurses:                     yes
   openssl:                     yes
   zlib:                        yes
   librsync:                    yes
   acl:                         no
   xattr:                       yes
   ipv6:                        yes


Cheers
Keith


On 30/07/15 10:16, Florent Peterschmitt wrote:

Forgot to ask: what burp version are you trying to build? Depending of that, librsync may be in a higher version like 1.x.

I haven't any OpenBSD box to try, but you could first try different versions of burp, like 1.3.x for example.

—
Reply to this email directly or view it on GitHub<https://github.com/grke/burp/issues/312#issuecomment-126236732>.


********************************************************************************************************************

This message may contain confidential information. If you are not the intended recipient please inform the
sender that you have received the message in error before deleting it.
Please do not disclose, copy or distribute information in this e-mail or take any action in reliance on its contents:
to do so is strictly prohibited and may be unlawful.

Thank you for your co-operation.

NHSmail is the secure email and directory service available for all NHS staff in England and Scotland
NHSmail is approved for exchanging patient data and other sensitive information with NHSmail and GSi recipients
NHSmail provides an email address for your career in the NHS and can be accessed anywhere

********************************************************************************************************************

--- a/configure.ac
+++ b/configure.ac
@@ -1445,35 +1445,6 @@ if test x$support_xattr = xyes -o x$support_xattr = xauto; then
       fi
    fi
 
-   dnl
-   dnl If we failed to find *BSD support and the Linux or OSX implementation of xattr try the Solaris xattr implementation
-   dnl
-   if test $have_xattr = no; then
-      AC_CHECK_HEADER(sys/attr.h, [ AC_DEFINE(HAVE_SYS_ATTR_H,1,[Defines if your system have the sys/attr.h header file])] , )
-      AC_CHECK_HEADER(sys/nvpair.h, [ AC_DEFINE(HAVE_SYS_NVPAIR_H,1,[Defines if your system have the sys/nvpair.h header file])] , )
-      AC_CHECK_HEADER(attr.h, [ AC_DEFINE(HAVE_ATTR_H,1,[Defines if your system have the attr.h header file])] , )
-
-      AC_CHECK_FUNCS(openat fstatat unlinkat fchownat futimesat,
-	 [
-	     have_xattr=yes
-	     AC_DEFINE([HAVE_OPENAT],1,[Define to 1 if you have the 'openat' function.])
-	     AC_DEFINE([HAVE_FSTATAT],1,[Define to 1 if you have the 'fstatat' function.])
-	     AC_DEFINE([HAVE_UNLINKAT],1,[Define to 1 if you have the 'unlinkat' function.])
-	     AC_DEFINE([HAVE_FCHOWNAT],1,[Define to 1 if you have the 'fchownat' function.])
-	     AC_DEFINE([HAVE_FUTIMESAT],1,[Define to 1 if you have the 'futimesat' function.])
-	 ]
-      )
-
-      if test $have_xattr = yes; then
-	 AC_CHECK_LIB(nvpair, nvlist_next_nvpair,
-	    [
-		AC_DEFINE([HAVE_NVLIST_NEXT_NVPAIR],1,[Define to 1 if you have the 'nvlist_next_nvpair' function.])
-		FDLIBS="-lnvpair $FDLIBS"
-	    ]
-	 )
-      fi
-   fi
-
    if test x$support_xattr = xyes -a $have_xattr != yes; then
       AC_MSG_ERROR([xattr support explicitly enabled but no supported xattr implementation found, 
   please either load the xattr libraries or rerun configure without --enable-xattr])

--- a/src/client/acl.c
+++ b/src/client/acl.c
@@ -10,7 +10,6 @@
 #ifdef HAVE_ACL
 #if defined(HAVE_LINUX_OS) || \
     defined(HAVE_FREEBSD_OS) || \
-    defined(HAVE_OPENBSD_OS) || \
     defined(HAVE_NETBSD_OS)
 #include "sys/acl.h"
 
@@ -25,7 +24,6 @@ static int acl_is_trivial(acl_t acl)
 {
 #if defined(HAVE_LINUX_OS) \
  || defined(HAVE_FREEBSD_OS) \
- || defined(HAVE_OPENBSD_OS) \
  || defined(HAVE_NETBSD_OS)
 	/*
 	 * acl is trivial if it has only the following entries:

--- a/src/client/acl.h
+++ b/src/client/acl.h
@@ -4,7 +4,6 @@
 #ifdef HAVE_ACL
 #if defined(HAVE_LINUX_OS) || \
     defined(HAVE_FREEBSD_OS) || \
-    defined(HAVE_OPENBSD_OS) || \
     defined(HAVE_NETBSD_OS)
 extern int has_acl(const char *path, enum cmd cmd);
 extern int get_acl(struct asfd *asfd, struct sbuf *sb,

--- a/src/client/extrameta.c
+++ b/src/client/extrameta.c
@@ -19,7 +19,6 @@ int has_extrameta(const char *path, enum cmd cmd, enum protocol protocol,
 	if(protocol==PROTO_2) return 0;
 #if defined(HAVE_LINUX_OS) || \
     defined(HAVE_FREEBSD_OS) || \
-    defined(HAVE_OPENBSD_OS) || \
     defined(HAVE_NETBSD_OS)
 
 #ifdef HAVE_ACL
@@ -28,7 +27,6 @@ int has_extrameta(const char *path, enum cmd cmd, enum protocol protocol,
 #endif
 #if defined(HAVE_LINUX_OS) || \
     defined(HAVE_FREEBSD_OS) || \
-    defined(HAVE_OPENBSD_OS) || \
     defined(HAVE_NETBSD_OS) || \
     defined(HAVE_DARWIN_OS)
 #ifdef HAVE_XATTR
@@ -52,7 +50,6 @@ int get_extrameta(struct asfd *asfd,
 	// some entries if acls were set.
 #if defined(HAVE_LINUX_OS) || \
     defined(HAVE_FREEBSD_OS) || \
-    defined(HAVE_OPENBSD_OS) || \
     defined(HAVE_NETBSD_OS)
 #ifdef HAVE_ACL
 	if(get_acl(asfd, sb, extrameta, elen, cntr)) return -1;
@@ -60,7 +57,6 @@ int get_extrameta(struct asfd *asfd,
 #endif
 #if defined(HAVE_LINUX_OS) || \
     defined(HAVE_FREEBSD_OS) || \
-    defined(HAVE_OPENBSD_OS) || \
     defined(HAVE_NETBSD_OS) || \
     defined(HAVE_DARWIN_OS)
 #ifdef HAVE_XATTR
@@ -115,7 +111,6 @@ int set_extrameta(struct asfd *asfd,
 #endif
 #if defined(HAVE_LINUX_OS) || \
     defined(HAVE_FREEBSD_OS) || \
-    defined(HAVE_OPENBSD_OS) || \
     defined(HAVE_NETBSD_OS)
 #ifdef HAVE_ACL
 			case META_ACCESS_ACL:
@@ -138,7 +133,6 @@ int set_extrameta(struct asfd *asfd,
 #endif
 #endif
 #if defined(HAVE_FREEBSD_OS) || \
-    defined(HAVE_OPENBSD_OS) || \
     defined(HAVE_NETBSD_OS)
 #ifdef HAVE_XATTR
 			case META_XATTR_BSD:

--- a/src/client/xattr.c
+++ b/src/client/xattr.c
@@ -12,7 +12,6 @@
 #if defined(HAVE_LINUX_OS) \
  || defined(HAVE_FREEBSD_OS) \
  || defined(HAVE_NETBSD_OS) \
- || defined(HAVE_OPENBSD_OS) \
  || defined(HAVE_DARWIN_OS)
 
 static char *get_next_str(struct asfd *asfd, char **data, size_t *l,
@@ -299,8 +298,7 @@ int set_xattr(struct asfd *asfd, const char *path,
 #endif // HAVE_LINUX_OS
 
 #if defined(HAVE_FREEBSD_OS) \
- || defined(HAVE_NETBSD_OS) \
- || defined(HAVE_OPENBSD_OS)
+ || defined(HAVE_NETBSD_OS)
 
 
 #include <sys/extattr.h>
@@ -583,6 +581,6 @@ int set_xattr(struct asfd *asfd, const char *path,
 	return -1;
 }
 
-#endif // HAVE_FREE/NET/OPENBSD_OS
+#endif // HAVE_FREE/NETBSD_OS
 
 #endif // HAVE_XATTR

--- a/src/client/xattr.h
+++ b/src/client/xattr.h
@@ -4,7 +4,6 @@
 #ifdef HAVE_XATTR
 #if defined(HAVE_LINUX_OS) \
  || defined(HAVE_FREEBSD_OS) \
- || defined(HAVE_OPENBSD_OS) \
  || defined(HAVE_NETBSD_OS) \
  || defined(HAVE_DARWIN_OS)
 extern int has_xattr(const char *path);

